lib/util/pthread_create_blocking_np.c

```diff
@@ -0,0 +1,168 @@
+#include <sys/time.h>
+
+#include <errno.h>
+#include <pthread.h>
+#include <stdlib.h>
+
+#include "pthread_create_blocking_np.h"
+
+struct wrapped_cookie {
+	/* User-supplied. */
+	void * (*start_routine)(void *);
+	void * arg;
+
+	/* Synchronization. */
+	pthread_mutex_t mutex;
+	pthread_cond_t cond;
+	int initialized;	/* 1 if initialized. */
```

Comment:
Rather than "initialized" -- which I think is a reference to the no-longer-present initialization callback? -- how about just "running"?